app/assets/javascripts/govuk_publishing_components/components/details.js

```diff
@@ -6,25 +6,31 @@ window.GOVUK.Modules.GovukDetails = window.GOVUKFrontend;
 
 (function (Modules) {
   function GemDetails ($module) {
+    console.log('details')
     this.$module = $module
-    this.customTrackLabel = this.$module.getAttribute('data-track-label')
+    this.$summary = this.$module.querySelector('.govuk-details__summary')
+
+    this.customTrackLabel = this.$summary.getAttribute('data-track-label')
     this.detailsClick = this.$module.querySelector('[data-details-track-click]')
+    console.log(this.customTrackLabel, this.detailsClick)
   }
 
   GemDetails.prototype.init = function () {
     if (this.customTrackLabel) { // If a custom label has been provided, we can simply call the tracking module
+      console.log('customTrackLabel')
       var trackDetails = new window.GOVUK.Modules.GemTrackClick()
-      trackDetails.start($(this.$module))
+      trackDetails.start($(this.$summary))
     } else if (this.detailsClick) { // If no custom label is set, we use the open/close status as the label
+      console.log('detailsClick')
       this.detailsClick.addEventListener('click', function (event) {
-        this.trackDefault(this.$module)
+        this.trackDefault(this.$summary)
       }.bind(this))
     }
   }
 
   GemDetails.prototype.trackDefault = function (element) {
     if (window.GOVUK.analytics && window.GOVUK.analytics.trackEvent) {
-      var componentStatus = (element.getAttribute('open') == null) ? 'open' : 'closed'
+      var componentStatus = (this.$module.getAttribute('open') == null) ? 'open' : 'closed'
```

Comment:
Welp, this appears to make no sense at first look but actually _does_ make sense because the `open` attribute is blank so returns `null` when not present and `""` when present... 🤯

Reply:
Yeah I didn't write this! However, things I learned today... the details element appends the 'open' attribute to itself by itself. Took me a few minutes to figure out how that was happening...